src/viam/components/arm/arm.py

```diff
@@ -112,3 +112,13 @@ async def is_moving(self) -> bool:
             bool: Whether the arm is moving.
         """
         ...
+
+    @abc.abstractmethod
+    async def do_command(self, command: Dict[str, Any], *, timeout: Optional[float] = None, **kwargs) -> Dict[str, Any]:
```

Comment:
This technically doesn't do anything, because the `ComponentBase` class (which all components inherit from) already has a `do_command` function that has a default implementation (it raises a `NotImplementedError` that you then have to override if you want to implement a `do`).

Reply:
This comment is applicable to all components

Reply:
Should we just remove `do_command` from `ComponentBase`?

Reply:
If we remove it from `ComponentBase`, this becomes a breaking change -- any existing custom components that rely on the default `NotImplemented` version of `do` would fail to load because they wouldn't implement all the abstract methods of this class.

Also, because *every* component should implement `do`, having it in `ComponentBase` means we reduce repetition.

And this allows modular components to opt-in to `do` easily, since all components (including modular ones) must inherit from `ComponentBase`. 

Those are just my reasons for keeping it in `ComponentBase`, but if anyone has strong feelings as to putting them in each individual component, I'm all ears